activerecord/lib/active_record/insert_all.rb

```diff
@@ -152,7 +152,7 @@ def find_unique_index_for(unique_by)
         name_or_columns = unique_by || model.primary_key
         match = Array(name_or_columns).map(&:to_s)
 
-        if index = unique_indexes.find { |i| match.include?(i.name) || i.columns == match }
+        if index = unique_indexes.find { |i| match.include?(i.name) || Array(i.columns).sort == match.sort }
```

Comment:
can you move the `match.sort` outside the loop? Likely not a big deal in most cases, but it bother me 😅

Reply:
Updated. 
So that's probably Shopify's scale taught to more carefully think about performance 😅

Reply:
Meh, the array is small, and it's very unlikely there's more than a dozen index on the table. It's just habit to never do unnecessary work however how small :).